File: src/main/java/io/github/mschout/gitlab/toggltimer/timer/TogglTimeEntry.java
package io.github.mschout.gitlab.toggltimer.timer;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.Instant;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import tools.jackson.databind.PropertyNamingStrategies;
import tools.jackson.databind.annotation.JsonNaming;

@Data
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@JsonInclude(JsonInclude.Include.NON_NULL)
@RequiredArgsConstructor
public class TogglTimeEntry {

  private final Long workspaceId;

  private final Long projectId;

  private final Instant start;

  private String description;

  private Long duration = -1L;

  private String createdWith = "Gitlab Toggl Timer";
}

File: src/main/kotlin/io/github/mschout/gitlab/toggltimer/timer/TogglTimeEntry.kt
package io.github.mschout.gitlab.toggltimer.timer

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.Instant
import tools.jackson.databind.PropertyNamingStrategies
import tools.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
@JsonInclude(JsonInclude.Include.NON_NULL)
data class TogglTimeEntry(
    val workspaceId: Long,
    val projectId: Long,
    val start: Instant,
    val description: String? = null,
    val duration: Long = -1L,
    val createdWith: String = "Gitlab Toggl Timer",
)
